compiler/luci/pass/src/helpers/LayerInfoMap.test.cpp

```diff
@@ -65,6 +120,54 @@ TEST(LayerInfoMapTest, simple_test)
   EXPECT_EQ(luci::QuantizationGranularity::ChannelWise, map["test"]->granularity);
 }
 
+TEST(LayerInfoMapTest, multiple_output_node_test)
+{
+  SplitAddTestGraph g;
+  g.init();
+
+  luci::LayerInfo info;
+  {
+    info.name = "split0";
+    info.dtype = loco::DataType::U8;
+    info.granularity = luci::QuantizationGranularity::ChannelWise;
+  }
+  std::vector<luci::LayerInfo> v;
+  v.emplace_back(info);
+  auto map = luci::layer_info_map(g.g(), v);
+
+  EXPECT_EQ(map.size(), 3);
+  EXPECT_EQ("split0", map["split"]->name);
+  EXPECT_EQ("split0", map["split0"]->name);
+  EXPECT_EQ("split0", map["split1"]->name);
```

Comment:
Hmm.. This behavior seems a bit weird. Could you make the behavior as below? I think that does not change the final quantized model.

```suggestion
  EXPECT_EQ("split", map["split"]->name);
  EXPECT_EQ("split0", map["split0"]->name);
  EXPECT_EQ("split1", map["split1"]->name);
```

Reply:
I agree with you, and I tried to solve this, but I found difficulties with it.
Actually, `map["some_name"]` returns `luci::LayerInfo *` and if initially there was no structure for the "some_name" (which is output node from mulitple output node in this example), then I see two solutions that will not lead to big changes. 

First, we add to the `map` for each out node name the current `LayerInfo` (which can contain name one of the output node, for example `split1`) and then we do not add anything to the original `std::vector<LayerInfo>` (this variant is current implementation).

The second way is to create a `LayerInfo` structure for each out node name and add to the `std::vector<LayerInfo>`, but at the same time I found that pointers can invalidate in the map because it keeps pointers to `luci::LayerInfo`.

Another way may be to create a structure for each output node (for multi output nodes) at the moment when the vector is created: https://github.com/Samsung/ONE/blob/76e9a6c1ef6498abde430ae279c9c0feda104daf/compiler/luci/pass/src/CircleQuantizer.cpp#L335

Reply:
The third option will add a search and check that the current name does not belong to multiple output node (or its output node), and so on for each name.

Reply:
Ah, it seems that `LayerInfoMap` contains inappropriate dtype. I didn't expect that new `LayerInfo` (not included in the `std::vector<LayerInfo>`) will be added.

I think it would be better to update `LayerInfoMap` to contain `LayerInfo`, instead of `LayerInfo *`. Then it'll be possible to add new `LayerInfo`s to the map.

Could you proceed after that change?

Reply:
I've made #8726 PTAL